cozy/structures/heaps.py

```diff
@@ -101,6 +111,8 @@ def typecheck(self, e : Exp, typecheck, report_err):
             ok = True
             if not (isinstance(e.e.type, TMinHeap) or isinstance(e.e.type, TMaxHeap)):
                 report_err(e, "cannot peek a non-heap")
+                ## TODO: Why is the `ok` variable needed at all?  Couldn't
+                ## you just return here and at the other assignment to `ok`?
```

Comment:
Not quite. We want both calls to `report_err` so that we can communicate all errors to the user rather than just stopping at the first one.